content/en/blog/_posts/2022-11-XX-runtime-observability/index.md

```diff
@@ -0,0 +1,275 @@
+enables us to combine logs, metrics, traces and other contextual information
+(called baggage) into a single resource. Cluster administrators or software
+engineers can use this resource to get a viewport about what is going on in the
+cluster over a defined period of time. But how can Kubernetes itself make use of
+this technology stack?
+
+[otel]: https://opentelemetry.io
+
+Kubernetes consists of multiple components where some are independent and others
+are stacked together. Looking at the architecture from a container runtime
+perspective, then we see from the top to the bottom:
+
+- **kube-apiserver**: Validates and configures data for the API objects
+- **kubelet**: Agent running on each node
+- **CRI runtime**: Container Runtime Interface (CRI) compatible container runtime
+  like [CRI-O][crio] or [containerd][containerd]
+- **OCI runtime**: Lower level [Open Container Initiative (OCI)][oci] runtime
+  like [runc][runc] or [crun][crun]
+- **Linux kernel**: Underlying operating system
+
+[crio]: https://cri-o.io
+[containerd]: https://containerd.io
+[oci]: https://opencontainers.org
+[runc]: https://github.com/opencontainers/runc
+[crun]: https://github.com/containers/crun
+
+That means if we encounter a problem with running containers in Kubernetes, then
+we start looking at one of those components. Finding the root cause for problems
+is one of the most time consuming actions we face with the increased
+architectural complexity from today's cluster setups. Even if we know the
+component which seems to cause the issue, we still have to take the others into
+account to maintain a mental timeline of events which are going on. How do we
+achieve that? Well, most folks will probably stick to scraping logs, filtering
+them and assembling them together over the components borders. We also have
+metrics, right? Correct, but bringing metrics values in correlation with plain
+logs makes it even harder to track what is going on. Some metrics are also not
+made for debugging purposes. They have been defined based on the end user
+perspective of the cluster for linking usable alerts and not for developers
+debugging a cluster setup.
+
+OpenTelemetry to the rescue, the project aims to combine signals like
+[traces][traces], [metrics][metrics] and [logs][logs] together to maintain the
+right viewport on the cluster state.
+
+[traces]: https://opentelemetry.io/docs/concepts/signals/traces
+[metrics]: https://opentelemetry.io/docs/concepts/signals/metrics
+[logs]: https://opentelemetry.io/docs/concepts/signals/logs
+
+What is the current state of OpenTelemetry tracing in Kubernetes? From an API
+server perspective, we have alpha support for tracing since Kubernetes v1.22,
+which will graduate to beta in one of the upcoming releases. Unfortunately the
+beta graduation has missed the v1.26 Kubernetes release. The design proposal can
+be found in the [corresponding Kubernetes Enhancement Proposal (KEP)][kep-647]
```

Comment:
(nit) It's nice to mention the title of KEPs when we link to them.

Reply:
Yep, done.